mat/matrix.go

```diff
@@ -35,7 +35,8 @@ type Matrix interface {
 // Dense types, especially helpful when adding new features.
 type allMatrix interface {
 	Reseter
-	IsZero() bool
+	IsEmpty() bool
+	Zero()
```

Comment:
Why was this added?

Reply:
Because all of the types implement it and it's relevant to the update?

Reply:
They all implement it, but it's not mentioned in the commit message.

Reply:
Done.

Reply:
It seems not entirely related, but I'll let it go.